File: fixtures/scale/java/large/src/cryptobench/ciphers/variantbeaufort/VariantBeaufortSuite.java
package cryptobench.ciphers.variantbeaufort;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class VariantBeaufortSuite implements CipherSuite {
    @Override
    public String name() {
        return "variantbeaufort";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new VariantBeaufortCipher(VariantBeaufortKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : VariantBeaufortVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/variantbeaufort/VariantBeaufortSuite.scala
package cryptobench.ciphers.variantbeaufort

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class VariantBeaufortSuite extends CipherSuite {
  override def name: String = "variantbeaufort"

  override def run(): SuiteResult = {
    val cipher: Cipher = new VariantBeaufortCipher(VariantBeaufortKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- VariantBeaufortVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
